protostar/testing/test_collector.py

```diff
@@ -275,12 +277,25 @@ def _build_test_suite_from_test_suite_info(
         test_suite_info: TestSuiteInfo,
     ) -> TestSuite:
         function_names = self._get_suite_function_names(test_suite_info.path)
-        setup_fn_name = self._collect_setup_hook_name(function_names)
+
+        if len(function_names) == 0:
+            names: List[str] = []
+            gas: List[AvailableGas] = []
+        elif isinstance(function_names[0], str):
+            function_names = cast(List[str], function_names)
+            names: List[str] = function_names
+            gas: List[AvailableGas] = [None] * len(function_names)
+        else:
+            function_names = cast(List[Tuple[str, AvailableGas]], function_names)
+            names: List[str] = [fn[0] for fn in function_names]
+            gas: List[AvailableGas] = [fn[1] for fn in function_names]
```

Comment:
```suggestion
        names: list[str] = []
        gas: list[AvailableGas] = []
        if len(function_names) == 0:
            pass
        elif isinstance(function_names[0], str):
            function_names = cast(list[str], function_names)
            names = function_names
            gas = [None] * len(function_names)
        else:
            function_names = cast(List[Tuple[str, AvailableGas]], function_names)
            names = [fn[0] for fn in function_names]
            gas = [fn[1] for fn in function_names]
```
This is only a quick idea, the point is that you declare types for the same variables multiple times which makes it worse for reading I think.

Reply:
I don't think there's that much of a difference

Reply:
With that you can actually omit the first `if`, it's not needed. Moreover, use `function_names` directly in the condition as an empty list is falsy

Reply:
Done